### PR TITLE
Remove centered

### PR DESCRIPTION
**Pull request by inailuig**
- removes QGTOnTheFly(..., centered=True)
- fixes a few imports & typos, unicode
- extends the tests of the logic to C->R
- drops an unnecessary cast of the output

### CODE REVIEW

**Comment by PhilipVinc** [comment]
can you link to the cast you removed?

File: netket/optimizer/qgt/qgt_onthefly_logic.py [hunk 1]
```diff
@@ -40,69 +38,27 @@ def O_vjp(forward_fn, params, samples, w):
...
     # The transposition of the 1D arrays is omitted in the implementation:
-    # (w^H O)^H -> (w* O)*
-
-    # TODO The allreduce in O_vjp could be deferred until after the tree_cast
-    # where the amount of data to be transferred would potentially be smaller
-    res = tree_conj(O_vjp(forward_fn, params, samples, w.conjugate()))
+    # (wᴴO)ᴴ -> (w* O)*
 
-    return tree_cast(res, params)
+    return tree_conj(O_vjp(forward_fn, params, samples, w.conjugate()))
```
Comment (inailuig) [inline netket/optimizer/qgt/qgt_onthefly_logic.py:51]: @PhilipVinc here

the vjp already returns the correct types

File: netket/optimizer/qgt/qgt_onthefly.py [hunk 2]
```diff
@@ -139,7 +130,7 @@ def onthefly_mat_treevec(
     else:
         ravel_result = False
 
-    vec = tree_cast(vec, S.params)
+    vec = nkjax.tree_cast(vec, S.params)
```
Comment (inailuig) [inline netket/optimizer/qgt/qgt_onthefly.py:134]: and here we should probably error/warn/implement it for anything other than the types of params
and not just silently cast. 
(same below)

Maybe in a separate PR though.

Comment (PhilipVinc) [reply netket/optimizer/qgt/qgt_onthefly.py:134]: Why should we error for different types of params?
I think it's still well defined, no?
If I convert the matrix to a dense matrix, then i can multiply it by a vector, whatever it's dtype.
Why can't i do it in the matmul then?

File: netket/optimizer/sr/deprecations.py [hunk 3]
```diff
@@ -16,6 +16,8 @@
 
 import jax
 
+import warnings
+
 from netket.utils import deprecated
```
Comment (PhilipVinc) [inline netket/optimizer/sr/deprecations.py:21]:
```suggestion
from netket.utils import deprecated, warn_deprecation
```

File: netket/optimizer/sr/deprecations.py [hunk 4]
```diff
@@ -36,11 +38,16 @@
     """
 )
 def SRLazyCG(diag_shift: float = 0.01, centered: bool = True, **kwargs):
```
Comment (PhilipVinc) [inline netket/optimizer/sr/deprecations.py:40]:
```suggestion
def SRLazyCG(diag_shift: float = 0.01, centered: bool = None, **kwargs):
```

File: netket/optimizer/sr/deprecations.py [hunk 5]
```diff
@@ -36,11 +38,16 @@
...
+
+    if centered:
+        warnings.warn(
+            """The argument `centered` is deprecated. The implementation now always behaves as if centered=False."""
+        )
```
Comment (PhilipVinc) [inline netket/optimizer/sr/deprecations.py:45]:
```suggestion
    if centered is not None:
        warn_deprecation(
            "The argument `centered` is deprecated. The implementation now always behaves as if centered=False."
        )
```

File: netket/optimizer/sr/deprecations.py [hunk 6]
```diff
@@ -59,10 +66,15 @@ def SRLazyCG(diag_shift: float = 0.01, centered: bool = True, **kwargs):
     """
 )
 def SRLazyGMRES(diag_shift: float = 0.01, centered: bool = True, **kwargs):
+
+    if centered:
+        warnings.warn(
+            """The argument `centered` is deprecated. The implementation now always behaves as if centered=False."""
+        )
```
Comment (PhilipVinc) [inline netket/optimizer/sr/deprecations.py:73]:
```suggestion
def SRLazyGMRES(diag_shift: float = 0.01, centered: bool = None, **kwargs):

    if centered is not None:
        warn_deprecation(
            "The argument `centered` is deprecated. The implementation now always behaves as if centered=False."
        )
```

File: netket/optimizer/qgt/qgt_onthefly.py [hunk 7]
```diff
@@ -40,9 +40,6 @@ def QGTOnTheFly(vstate=None, **kwargs) -> "QGTOnTheFlyT":
 
     Args:
         vstate: The variational State.
-        centered: Uses S=⟨ΔÔᶜΔÔ⟩ if True (default), S=⟨ÔᶜΔÔ⟩ otherwise. The two forms are
-            mathematically equivalent, but might lead to different results due to numerical
-            precision. The non-centered variant should be approximately 33% faster.
     """
     if vstate is None:
         return partial(QGTOnTheFly, **kwargs)
```
Comment (PhilipVinc) [inline netket/optimizer/qgt/qgt_onthefly.py:45]:
```suggestion
        return partial(QGTOnTheFly, **kwargs)
     
    if "centered" in kwargs:
        warn_deprecated(
            "The argument `centered` is deprecated. The implementation now always behaves as if centered=False."
            )
```

**Review by PhilipVinc — APPROVED** [review]
with those changes it will be ok

**Pull request merged**